test/test-context.rb

```diff
@@ -132,6 +132,33 @@ def test_match_escalation_threshold
     assert_equal(-1, context.match_escalation_threshold)
   end
 
+  sub_test_case(".match_escalation?") do
+    def setup
+      default_match_escalation =
+        context.match_escalation?
+      begin
+        yield
+      ensure
+        context.match_escalation =
+          default_match_escalation
+      end
+    end
+
+    def test_true
+      context.match_escalation = true
+      assert do
+        context.match_escalation?
+      end
```

Comment:
It seems that `assert_true` is good enough instead of `assert do`

Reply:
`assert do` is better for predicate.